File: BackEnd/event-service/src/main/java/it/profice/project/event_service/model/Event.java
package it.profice.project.event_service.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.*;

import java.util.Date;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Event {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String uuid;
    private String title;
    private String description;
    private Date date;
    private String location;
    private Integer totalSeats;
    private Integer availableSeats;
    private boolean state;
    private Category eventCategory;
}

File: FrontEnd/EventBazaar/src/app/model/event.ts
export interface Event {
    uuid: string;
    title: string;
    description: string;
    date: Date;
    location: string;
    totalSeats: number;
    availableSeats: number;
    state: false;
    eventCategory: EventCategory;
}
export const defaultBeer: Event = {
    uuid: '',
    title: '',
    description: '',
    date: new Date('2025-01-01'),
    location: '',
    totalSeats: 0,
    availableSeats: 0,
    state: false,
    eventCategory: 0
}

export enum EventCategory {
    CONCERTS,
    PLAY,
    SPORT,
    EXHIBITIONS,
    OTHER
}